cns/service/main.go

```diff
@@ -246,6 +246,27 @@ var args = acn.ArgumentList{
 	},
 }
 
+func init() {
```

Comment:
why `init()` function instead of main? I tend to avoid init when possible since it can cause some confusion, but if there is a compelling reason, I'm not necessarily against it

Reply:
doing this type of work in an init is more canonical than main.
initializing globals that need to exist before any code in the package runs is the stated intent of init functions. this stuff could be at the top of main, but that does less to communicate that it _must_ be done before anything in the package runs, and there could be other entry points in to code in this package than the main func (obviously there _aren't_, but theoretically).

Reply:
Where is this called? Is it called before main()? I agree with Matt, it gets confusing (at least for the readers coming from non-go world). We havent used this pattern in other services. May be a detailed comment in both init and main function will future developers to follow this standard

Reply:
`init()`s are called automatically when the package is imported - before any code runs in them runs. they are used for things like building pre-run state that can't be done in package level declarations. for example, here, I want to set up the signal handling before any code runs so that the program can exit gracefully even before `main()` is invoked.
more info: https://golang.org/doc/effective_go#init
spec: https://golang.org/ref/spec#Package_initialization

Reply:
Looks like this is the intended usage of `init()` according to the official style guide, so I'm happy with it this way (in fact I prefer to follow the style guide, so wouldn't be happy any other way)

Reply:
Sounds good, Evan can we add a function comment to Init() and probably link the go guideline